### PR TITLE
Integrating riensen/Auto-GPT-Email-Plugin

### PR DESCRIPTION
**Pull request by riensen**
This PR migrates my repo github.com/riensen/Auto-GPT-Email-Plugin

### CODE REVIEW

File: src/autogpt_plugins/email/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Auto-GPT Email Plugin: Revolutionize Your Email Management with Auto-GPT 🚀
+
+The Auto-GPT Email Plugin is an innovative and powerful plugin for the groundbreaking base software, Auto-GPT. Harnessing the capabilities of the latest Auto-GPT architecture, Auto-GPT aims to autonomously achieve any goal you set, pushing the boundaries of what is possible with artificial intelligence. This email plugin takes Auto-GPT to the next level by enabling it to send and read emails, opening up a world of exciting use cases.
+
+[![GitHub Repo stars](https://img.shields.io/github/stars/riensen/Auto-GPT-Email-Plugin?style=social)](https://github.com/riensen/Auto-GPT-Email-Plugin/stargazers)
```
Comment (ntindle) [inline src/autogpt_plugins/email/README.md:5]: may want to repoint this to the main repo?

File: src/autogpt_plugins/email/__init__.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    role: str
+    content: str
+
+
+class AutoGPTEmailPlugin(AutoGPTPluginTemplate):
+    """
+    This is the Auto-GPT email plugin.
+    """
+
+    def __init__(self):
```
Comment (ntindle) [inline src/autogpt_plugins/email/__init__.py:18]: We should check if your required envs exist before we register

**Review by ntindle — CHANGES_REQUESTED** [review]

File: src/autogpt_plugins/email/__init__.py (hunk 2 above)
Comment (riensen) [reply src/autogpt_plugins/email/__init__.py:18]: What are you trying to achieve? Do you want to throw an exception if the env vars do not exist?

Comment (ntindle) [reply src/autogpt_plugins/email/__init__.py:18]: I'd prefer it to not crash if they don't exist for all plugins

Comment (riensen) [reply src/autogpt_plugins/email/__init__.py:18]: So in this case, it is working as intended right now. If they are not set, then it won't crash right now.
As opposed to the Twitter plugin. Here, if the env vars are not set, then it crashes with an exception.

Comment (riensen) [reply src/autogpt_plugins/email/__init__.py:18]: What we could do is not register the commands if the minimum required env vars are not set.

File: src/autogpt_plugins/email/README.md (hunk 1 above)
Comment (riensen) [reply src/autogpt_plugins/email/README.md]: done

**Review by ntindle — APPROVED** [review]

**Comment by ntindle** [comment]
Fix the readme and we should be GTG here

**Review by ntindle — APPROVED** [review]

**Pull request merged**